src/sidebar/components/test/login-control-test.js

```diff
@@ -2,350 +2,83 @@
-        });
-
-        it('shows the enabled user profile button', function() {
-          assert.isTrue(isUserProfileButtonEnabled(thirdPartyUserPage()));
-        });
-
-        it('sends the PROFILE_REQUESTED event', function() {
-          thirdPartyUserPage().userProfileButton.click();
-
-          assert.calledOnce(fakeBridge.call);
-          assert.calledWithExactly(
-            fakeBridge.call,
-            bridgeEvents.PROFILE_REQUESTED
-          );
-        });
-
-        it('does not open a new tab', function() {
-          thirdPartyUserPage().userProfileButton.click();
-
-          assert.notCalled(fakeWindow.open);
-        });
-      });
-    });
-  });
-
-  describe('the account settings button', function() {
-    context('when the user auth status is unknown', function() {
-      it('does not show', function() {
-        assert.isNull(unknownAuthStatusPage().accountSettingsButton);
-      });
-    });
-
-    context('when the user is logged out', function() {
-      it('does not show', function() {
-        assert.isNull(loggedOutPage().accountSettingsButton);
-      });
-    });
-
-    context('when a first-party user is logged in', function() {
-      it('does show', function() {
-        assert.isNotNull(firstPartyUserPage().accountSettingsButton);
-      });
-    });
-
-    context('when a third-party user is logged in', function() {
-      it('does not show', function() {
-        assert.isNull(thirdPartyUserPage().accountSettingsButton);
-      });
-    });
-  });
-
-  describe('the help button', function() {
-    context('when the user auth status is unknown', function() {
-      it('does show', function() {
-        assert.isNotNull(unknownAuthStatusPage().helpButton);
-      });
-    });
-
-    context('when the user is logged out', function() {
-      it('does show', function() {
-        assert.isNotNull(loggedOutPage().helpButton);
-      });
-    });
-
-    context('when a first-party user is logged in', function() {
-      it('does show', function() {
-        assert.isNotNull(firstPartyUserPage().helpButton);
-      });
-    });
-
-    context('when a third-party user is logged in', function() {
-      it('does show', function() {
-        assert.isNotNull(thirdPartyUserPage().helpButton);
-      });
-    });
+    angular.mock.module('app', {});
   });
 
-  describe('the log out button', function() {
-    context('when the user auth status is unknown', function() {
-      it('does not show', function() {
-        assert.isNull(unknownAuthStatusPage().logOutButton);
-      });
-    });
-
-    context('when the user is logged out', function() {
-      it('does not show', function() {
-        assert.isNull(loggedOutPage().logOutButton);
-      });
-    });
-
-    context('when a first-party user is logged in', function() {
-      it('does show', function() {
-        assert.isNotNull(firstPartyUserPage().logOutButton);
+  context('sign up and log in links', () => {
```

Comment:
These tests are fairly facile and lightweight; happy to approach differently if there is a better way.

Reply:
In the client at least, `context` is usually used to describe a _condition_ (eg. "_when_ the foobar setting is enabled") and `describe` is used to refer to _what_ is being tested (either a specific function or method, or a terse description of the UI elements or functionality)